macos_proxy_notarization/proxy_notarization.sh

```diff
@@ -0,0 +1,134 @@
+
+  echo "Unlock the keychain 2"
+  ls
+  echo $CERTIFICATE_P12
+  echo $ESO_TEAM_P12
+  security import ./eso_certificate.p12 -x -t agg -k $KEY_CHAIN -P $WAVEFRONT_TEAM_CERT_PASSWORD -T /usr/bin/codesign;
+  security import ./certificate.p12 -x -t agg -k $KEY_CHAIN -P $CERTIFICATE_PASSWORD -T /usr/bin/codesign;
+
+  echo "Finding identity"
+  security find-identity -v
+
+  echo "Unlock the keychain 3"
+  security set-key-partition-list -S apple-tool:,apple:,codesign: -s -k travis $KEY_CHAIN
+
+  echo "Delete certs"
+  rm -fr *.p12
+}
+
+# Parse the proxy version our of the 
+parse_proxy_version_tarball() {
+  echo "Get the version"
+  TO_BE_NOTARIZED=$(aws s3 ls s3://eso-wfproxy-testing/to_be_notarized/$WFPROXY_TARBALL | awk '{print $4}')
+  RE=[0-9]+\.[0-9]+\.[0-9]+
+  if [[ $TO_BE_NOTARIZED =~ $RE ]]; then 
+    echo ${BASH_REMATCH[0]};
+    VERSION=${BASH_REMATCH[0]} 
+  fi
+  echo $VERSION
+}
+
+# CP non-notarized proxy, cp signed jdk with it, package as .zip
+repackage_proxy() {
+  COPY_FORM_TO_BE_NOTARIZED="aws s3 cp s3://eso-wfproxy-testing/to_be_notarized/wfproxy-$VERSION.tar.gz ."
+  $COPY_FORM_TO_BE_NOTARIZED
+  TARBALL="wfproxy-$VERSION.tar.gz"
+  tar xvzf $TARBALL
+  zip -r wavefront-proxy-$VERSION.zip bin/ etc/ lib/
+}
+
+# Notarized the .zip and upload to Apply
+notarized_newly_package_proxy() {
+  echo "Codesigning the wavefront-proxy package"
+  codesign -f -s "$ESO_DEV_ACCOUNT" wavefront-proxy-$VERSION.zip --deep --options runtime
+
+  echo "Verifying the codesign"
+  codesign -vvv --deep --strict wavefront-proxy-$VERSION.zip
+
+  echo "Uploading the package for Notarization"
+  response="$(xcrun altool --notarize-app --primary-bundle-id "com.wavefront" --username "$USERNAME" --password "$APP_SPECIFIC_PW" --file "wavefront-proxy-$VERSION.zip" | sed -n '2 p')"
+  echo $response
+
+  echo "Grabbing Request UUID"
+  requestuuid=${response#*= }
+  echo $requestuuid
+
+  echo "Executing this command to see the status of notarization"
+  xcrun altool --notarization-info "$requestuuid" -u "$USERNAME" -p "$APP_SPECIFIC_PW"
+}
+
+# Pass or fail based on notarization status
+wait_for_notarization() {
+  status="$(xcrun altool --notarization-info "$requestuuid" -u "$USERNAME" -p "$APP_SPECIFIC_PW")"
+  in_progress='Status: in progress'
+  success='Status Message: Package Approved'
+  invalid='Status: invalid'
+
+  while true;
+  do
+    echo $status
+    if [[ "$status" == *"$success"* ]]; then
+      echo "Successful notarization"
+      aws s3 cp wavefront-proxy-$VERSION.zip s3://wavefront-cdn/brew/
+      exit 0
+    elif [[ "$status" == *"$in_progress"* ]]; then
+      status="$(xcrun altool --notarization-info "$requestuuid" -u "$USERNAME" -p "$APP_SPECIFIC_PW")"
+      sleep 60
+    elif [[ "$status" == *"$invalid"* ]]; then
+      echo "Failed notarization"
+      exit 1
+    fi
+  done
+}
+
+main() {
+  check_notarized_list
+  create_dev_certs
+  parse_proxy_version_tarball
+  echo $VERSION
```

Comment:
are we not passing the version as param from jenkins job?

Reply:
I'm passing in the tarball file name ie: "wfproxy-7.0.0.tar.gz" 
the variable, "VERSION" is the parsing the actual version number:7.0.0